File: common/src/test/scala/com/taskadapter/connector/common/MappingBuilderTest.scala
package com.taskadapter.connector.common

import com.taskadapter.connector.definition.{ExportDirection, FieldMapping}
import com.taskadapter.connector.MappingBuilder
import com.taskadapter.model.Field
import org.junit.runner.RunWith
import org.scalatest.junit.JUnitRunner
import org.scalatest.{FunSpec, Matchers}

@RunWith(classOf[JUnitRunner])
class MappingBuilderTest extends FunSpec with Matchers {
  it("skips fields that are not selected") {
    val rows = MappingBuilder.build(
      List(FieldMapping(Field("summary"), Field("summary"), false, "default")),
      ExportDirection.RIGHT

    )
    rows shouldBe empty
  }

  it("export Right processes selected fields") {
    val rows = MappingBuilder.build(
      List(FieldMapping(Field("JiraSummary"), Field("RedmineSummary"), true, "default")),
      ExportDirection.RIGHT

    )
    rows.head.sourceField.get.name shouldBe "JiraSummary"
    rows.head.targetField.get.name shouldBe "RedmineSummary"
  }

  it("export Left processes selected fields") {
    val rows = MappingBuilder.build(
      List(FieldMapping(Field("JiraSummary"), Field("RedmineSummary"), true, "default")),
      ExportDirection.LEFT

    )
    rows.head.sourceField.get.name shouldBe "RedmineSummary"
    rows.head.targetField.get.name shouldBe "JiraSummary"
  }

}

File: webshared/src/test/java/com/taskadapter/common/ui/MappingBuilderTest.java
package com.taskadapter.common.ui;

import com.taskadapter.connector.FieldRow;
import com.taskadapter.connector.definition.ExportDirection;
import com.taskadapter.model.CustomString;
import org.junit.Test;

import java.util.List;

import static org.assertj.core.api.Assertions.assertThat;

public class MappingBuilderTest {
    @Test
    public void skipsFieldsThatAreNotSelected() {
        var rows = MappingBuilder.build(
                List.of(FieldMapping.apply(new CustomString("summary"), new CustomString("summary"), false, "default")),
                ExportDirection.RIGHT

        );
        assertThat(rows).isEmpty();
    }

    @Test
    public void exportRightProcessesSelectedFields() {
        List<FieldRow<?>> rows = MappingBuilder.build(
                List.of(FieldMapping.apply(
                        new CustomString("JiraSummary"), new CustomString("RedmineSummary"), true, "default")),
                ExportDirection.RIGHT

        );
        assertThat(rows.get(0).getSourceField().get().getFieldName()).isEqualTo("JiraSummary");
        assertThat(rows.get(0).getTargetField().get().getFieldName()).isEqualTo("RedmineSummary");
    }

    @Test
    public void exportLeftProcessesSelectedFields() {
        List<FieldRow<?>> rows = MappingBuilder.build(
                List.of(FieldMapping.apply(
                        new CustomString("JiraSummary"), new CustomString("RedmineSummary"), true, "default")),
                ExportDirection.LEFT

        );
        assertThat(rows.get(0).getSourceField().get().getFieldName()).isEqualTo("RedmineSummary");
        assertThat(rows.get(0).getTargetField().get().getFieldName()).isEqualTo("JiraSummary");
    }
}
